File: Laboratory-works-1-course-2-semester/Lab2_cp/StringProcessor.cpp
#include "StringProcessor.h"
#include <iostream>

StringProcessor::StringProcessor() : str("") {}
StringProcessor::StringProcessor(const std::string& inputStr) : str(inputStr) {}
StringProcessor::StringProcessor(const StringProcessor& other) : str(other.str) {}

StringProcessor::~StringProcessor() 
{
    std::cout << "Объект уничтожен. Строка: " << str << std::endl;
}

size_t StringProcessor::getLength() const 
{
    return str.length();
}

void StringProcessor::removeFives() 
{
    str.erase(remove(str.begin(), str.end(), '5'), str.end());
}

std::string StringProcessor::getString() const 
{
    return str;
}

File: Laboratory-works-1-course-2-semester/Lab2CS/StringProcessor.cs
using System;

public class StringProcessor
{
    private string str;

    public StringProcessor()
    {
        str = "";
    }

    public StringProcessor(string inputStr)
    {
        str = inputStr;
    }

    public StringProcessor(StringProcessor other)
    {
        str = other.str;
    }

    ~StringProcessor()
    {
        Console.WriteLine("Объект уничтожен. Строка: " + str);
    }

    public int GetLength()
    {
        return str.Length;
    }

    public void RemoveFives()
    {
        str = str.Replace("5", "");
    }

    public string GetString()
    {
        return str;
    }
}
